ipamutils/utils.go

```diff
@@ -34,19 +33,47 @@ type NetworkToSplit struct {
 	Size int    `json:"size"`
 }
 
-// InitNetworks initializes the broad network pool and the granular network pool
-func InitNetworks(defaultAddressPool []*NetworkToSplit) {
-	initNetworksOnce.Do(func() {
-		// error ingnored should never fail
-		PredefinedGranularNetworks, _ = splitNetworks(defaultGranularNetwork)
-		if defaultAddressPool == nil {
-			defaultAddressPool = defaultBroadNetwork
-		}
-		var err error
-		if PredefinedBroadNetworks, err = splitNetworks(defaultAddressPool); err != nil {
-			logrus.WithError(err).Error("InitAddressPools failed to initialize the default address pool")
-		}
-	})
+func init() {
+	var err error
+	if PredefinedGlobalScopeDefaultNetworks, err = splitNetworks(globalScopeDefaultNetworks); err != nil {
+		logrus.WithError(err).Error("InitAddressPools failed to initialize the global scope default address pool")
+	}
+	if PredefinedLocalScopeDefaultNetworks, err = splitNetworks(localScopeDefaultNetworks); err != nil {
+		logrus.WithError(err).Error("InitAddressPools failed to initialize the local scope default address pool")
+	}
+}
+
+// configDefaultNetworks configures local as well global default pool based on input
+func configDefaultNetworks(defaultAddressPool []*NetworkToSplit, result *[]*net.IPNet) error {
+	defaultNetworks, err := splitNetworks(defaultAddressPool)
+	if err != nil {
+		return err
+	}
+	*result = defaultNetworks
+	return nil
+}
+
+// ConfigGlobalScopeDefaultNetworks configures global default pool.
+// Ideally this will be called from SwarmKit as part of swarm init
+func ConfigGlobalScopeDefaultNetworks(defaultAddressPool []*NetworkToSplit) error {
+	defer mutex.Unlock()
+
+	if defaultAddressPool == nil {
+		defaultAddressPool = globalScopeDefaultNetworks
+	}
```

Comment:
Since the globalScopeDefaultNetworks are already installed, why not just do:
```
if defaultAddressPool == nil {
    return nil
}
```
at the start of the function?  (even before the lock/deferred unlock) Re-invoking configureDefaultNetworks() again on globalScopeDefaultNetworks isn't going to change the configuration.

Reply:
No. This is one of the issue I found out during testing and added as extra requirement after I & @mark-church discussed.
here is the scenario which I am trying to address 
1) swarm init happens with new "subnet-pool"
2) tearing down swarm cluster
3) now bring up swarm init without "subnet-pool" option
4) we should init global pool back to inbuilt pool as it used to work without this feature.
so I will have to set it back to in built values and re initialize it again.  else swam cluster will have old IPs for overlay networks.

Reply:
ok, got it. leave as is, then.